clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClient.java

```diff
@@ -187,7 +187,10 @@ private <HttpT, RespT> void sendRequest(
         final HttpEntity entity = (HttpEntity) body;
         final byte[] entityBytes;
         try {
-          entityBytes = EntityUtils.toByteArray(entity);
+          final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+          entity.writeTo(byteArrayOutputStream);
+          entityBytes = byteArrayOutputStream.toByteArray();
+          byteArrayOutputStream.close();
         } catch (final IOException e) {
```

Comment:
🔧 `ByteArrayOutputStream` is an auto-closable, so it should be safer to write a try-with-resources statement.
```suggestion
        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
          entityBytes = EntityUtils.toByteArray(entity);
          entity.writeTo(byteArrayOutputStream);
          entityBytes = byteArrayOutputStream.toByteArray();
        } catch (final IOException e) {
```